govcd/vm_test.go

```diff
@@ -810,3 +810,101 @@ func (vcd *TestVCD) Test_VMPowerOnPowerOff(check *C) {
 	check.Assert(err, IsNil)
 	check.Assert(vmStatus, Equals, "POWERED_OFF")
 }
+
+func (vcd *TestVCD) Test_GetNetworkConnectionSection(check *C) {
+	if vcd.skipVappTests {
+		check.Skip("Skipping test because vapp was not successfully created at setup")
+	}
+
+	vapp := vcd.findFirstVapp()
+	vmType, vmName := vcd.findFirstVm(vapp)
+	if vmName == "" {
+		check.Skip("skipping test because no VM is found")
+	}
+
+	vm, err := vcd.client.Client.FindVMByHREF(vmType.HREF)
+	check.Assert(err, IsNil)
+
+	networkBefore, err := vm.GetNetworkConnectionSection()
+	check.Assert(err, IsNil)
+	//check.Assert(network, IsNil)
+
+	err = vm.UpdateNetworkConnectionSection(networkBefore)
+	check.Assert(err, IsNil)
+
+	networkAfter, err := vm.GetNetworkConnectionSection()
+	check.Assert(err, IsNil)
+
+	// Filter out always differing fields and do deep comparison of objects
+	networkBefore.Link = &types.Link{}
+	networkAfter.Link = &types.Link{}
+	check.Assert(networkAfter, DeepEquals, networkBefore)
+
+}
+
+// Test_PowerOnAndForceCustomization uses the VM from TestSuite and forces guest customization
+// in addition to the one which is triggered on first boot. It waits until the initial guest
+// customization after first power on is finished because it is inherited from the template.
+// After this initial wait it Undeploys VM and triggers a second customization and again waits until guest
+// customization status exits "GC_PENDING" state to succeed the test.
+// This test relies on longer timeouts in BlockWhileGuestCustomizationStatus because VMs take a lengthy time
+// to boot up and report customization done.
+func (vcd *TestVCD) Test_PowerOnAndForceCustomization(check *C) {
+	if vcd.skipVappTests {
+		check.Skip("Skipping test because vApp wasn't properly created")
+	}
+
+	fmt.Printf("Running: %s\n", check.TestName())
+	vapp := vcd.findFirstVapp()
+	vmType, vmName := vcd.findFirstVm(vapp)
+	if vmName == "" {
+		check.Skip("skipping test because no VM is found")
+	}
+
+	vm, err := vcd.client.Client.FindVMByHREF(vmType.HREF)
+	check.Assert(err, IsNil)
+
+	// It may be that prebuilt VM was not booted before in the test vApp and it would still have
+	// a guest customization status 'GC_PENDING'. This is because initially VM has this flag set
+	// but while this flag is here the test cannot actually check if vm.PowerOnAndForceCustomization()
+	// gives any effect therefore we must "wait through" initial guest customization if it is in
+	// 'GC_PENDING' state.
+	custStatus, err := vm.GetGuestCustomizationStatus()
+	check.Assert(err, IsNil)
+	if custStatus == "GC_PENDING" {
```

Comment:
should `CG_PENDING` be an item in `types.VAppStatuses`?

Reply:
No, it is not vApp status. It is VM guest customization status. It can be constants on their own. Added known statuses to constants.